### PR TITLE
[GlobalISel] Import samesign flag

### PR DESCRIPTION
**Pull request by tschuett**
Credits: https://github.com/llvm/llvm-project/pull/111419

### CODE REVIEW

File: llvm/lib/CodeGen/MIRParser/MIParser.cpp [hunk 1]
```diff
@@ -1476,7 +1476,8 @@ bool MIParser::parseInstruction(unsigned &OpCode, unsigned &Flags) {
          Token.is(MIToken::kw_noconvergent) ||
          Token.is(MIToken::kw_unpredictable) ||
          Token.is(MIToken::kw_nneg) ||
-         Token.is(MIToken::kw_disjoint)) {
+         Token.is(MIToken::kw_disjoint) ||
+         Token.is(MIToken::kw_samesign)) {
```
Comment (arsenm) [inline llvm/lib/CodeGen/MIRParser/MIParser.cpp:1480]: Missing MIRParser test

File: llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign.ll [hunk 2]
```diff
@@ -0,0 +1,69 @@
+; NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 4
+; RUN: llc -mtriple=aarch64-linux-gnu -O0 -stop-after=irtranslator -global-isel -verify-machineinstrs %s -o - 2>&1 | FileCheck %s
+
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign.ll:3]:
```suggestion
; RUN: llc -global-isel -mtriple=aarch64-linux-gnu -O0 -stop-after=irtranslator < %s | FileCheck %s
```

File: llvm/lib/CodeGen/MIRParser/MIParser.cpp (hunk 1 above)
Comment (tschuett) [reply llvm/lib/CodeGen/MIRParser/MIParser.cpp:1480]: Can you give me a pointer? The last one only had the import test:
https://github.com/llvm/llvm-project/pull/86650/files

Comment (arsenm) [reply llvm/lib/CodeGen/MIRParser/MIParser.cpp:1480]: llvm/test/CodeGen/MIR/X86/fastmath.mir, should expand this to cover all the mir flags

File: llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign-mir.mir [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+name:            icmp_samesign
+body:             |
+  bb.0:
+    liveins: $w0, $w1
+
+    ; CHECK-LABEL: name: icmp_samesign
+    ; CHECK: liveins: $w0, $w1
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: %y:_(s32) = COPY $w1
+    ; CHECK-NEXT: %cmp:_(s1) = samesign G_ICMP intpred(eq), %y(s32), %y
```
Comment (tschuett) [inline llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign-mir.mir:15]: insufficient?

File: llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign-mir.mir [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+# RUN: llc -mtriple aarch64 -run-pass=aarch64-prelegalizer-combiner -verify-machineinstrs %s -o - | FileCheck %s
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign-mir.mir:2]: Test name says IRTranslator but this is not testing the irtranslator

Comment (tschuett) [reply llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign-mir.mir:2]: It is part of the import of the samesign flag. Suggestions for better name?

Comment (arsenm) [reply llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign-mir.mir:2]: If there's really no combine being tested here, move to llvm/test/CodeGen/MIR/ and name icmp-flags?

**Comment by tschuett** [comment]
Ping.

File: llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign.ll [hunk 5]
```diff
@@ -0,0 +1,69 @@
+; NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 4
+; RUN: llc -global-isel -mtriple=aarch64-linux-gnu -O0 -stop-after=irtranslator < %s | FileCheck %s
+
+
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/GlobalISel/irtranslator-samesign.ll:4]:
```suggestion
```

**Review by arsenm — APPROVED** [review]

File: llvm/test/CodeGen/MIR/icmp-flags.mir [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    ; CHECK-NEXT: %zext:_(s32) = G_ZEXT %cmp(s1)
+    ; CHECK-NEXT: $w0 = COPY %zext(s32)
+    ; CHECK-NEXT: RET_ReallyLR implicit $w0
+    %x:_(s32) = COPY $w0
+    %y:_(s32) = COPY $w1
+    %cmp:_(s1) = G_ICMP intpred(eq), %y:_(s32), %y:_
+    %zext:_(s32) = G_ZEXT %cmp:_(s1)
+    $w0 = COPY %zext
+    RET_ReallyLR implicit $w0
+---
```
Comment (antoniofrighetto) [inline llvm/test/CodeGen/MIR/icmp-flags.mir:50]: This test needs to go under `MIR/AArch64`.

**Pull request merged**